src/main/java/org/elasticsearch/cloud/aws/blobstore/S3BlobStore.java

```diff
@@ -71,6 +74,18 @@ public S3BlobStore(Settings settings, AmazonS3 client, String bucket, @Nullable
         }
 
         this.numberOfRetries = maxRetries;
+
+        try{
```

Comment:
Do you know which exception would be thrown if no permission was there? Would make the logging better. Ignore if hard.